docs/user/guides/custom-domains.rst

```diff
@@ -0,0 +1,120 @@
+============================
+
+This guide describes how to host your documentation using your own domain name, such as ``docs.example.com``.
+
+.. contents:: Contents
+    :local:
+
+Adding a custom domain
+----------------------
+
+To setup your :doc:`custom domain </custom-domains>`, follow these steps:
+
+#. Go the :guilabel:`Admin` tab of your project.
+#. Click on :guilabel:`Domains`.
+#. Enter the domain where you want to serve the documentation from (e.g. ``docs.example.com``).
+#. Mark the :guilabel:`Canonical` option if you want use this domain
+   as your :doc:`canonical domain </canonical-urls>`.
+#. Click on :guilabel:`Add`.
+#. At the top of the next page you'll find the value of the DNS record that you need to point your domain to.
+   For |org_brand| this is ``readthedocs.io``, and for :doc:`/commercial/index`
+   the record is in the form of ``<hash>.domains.readthedocs.com``.
+
+   .. note::
+
+      For a subdomain like ``docs.example.com`` add a CNAME record,
+      and for a root domain like ``example.com`` use an ANAME or ALIAS record.
+
+We provide a validated SSL certificate for the domain,
+managed by `Cloudflare <https://www.cloudflare.com/>`_.
+The SSL certificate issuance should happen within a few minutes,
+but might take up to one hour.
+See `SSL certificate issue delays`_ for more troubleshooting options.
+
+To see if your DNS change has propagated, you can use a tool like ``dig`` to inspect your domain from your command line.
+As an example, our blog's DNS record looks like this:
+
+.. prompt:: bash $, auto
+
+   $ dig +short CNAME blog.readthedocs.com
+     readthedocs.io.
+
+.. warning::
+
+   We don't support pointing subdomains or root domains to a project using A records.
+   DNS A records require a static IP address and our IPs may change without notice.
+
+
+Removing a custom domain
+------------------------
+
+To remove a custom domain:
+
+#. Go the :guilabel:`Admin` tab of your project.
+#. Click on :guilabel:`Domains`.
+#. Click the :guilabel:`Remove` button next to the domain.
+#. Click :guilabel:`Confirm` on the confirmation page.
+
+.. warning::
+
+    Once a domain is removed,
+    your previous documentation domain is no longer served by Read the Docs,
+    and any request for it will return a 404 Not Found!
+
+Strict Transport Security (HSTS) and other custom headers
+---------------------------------------------------------
+
+By default, we do not return a `Strict Transport Security header`_ (HSTS) for user custom domains.
+This is a conscious decision as it can be misconfigured in a not easily reversible way.
+For both |org_brand| and |com_brand|, HSTS and other custom headers can be set upon request.
+
+We always return the HSTS header with a max-age of at least one year
+for our own domains including ``*.readthedocs.io``, ``*.readthedocs-hosted.com``, ``readthedocs.org`` and ``readthedocs.com``.
+
+Please contact :doc:`/support` if you want to add a custom header to your domain.
```

Comment:
Note: It's because of this exact instruction that I think we can keep this under How-To. There isn't much to be said in an Explanation on this topic. If someone needs HSTS, they know what they are doing and can probably find it natural that this section is in the How-To.